observability/opentelemetry-extension/src/main/java/org/wildfly/extension/opentelemetry/extension/OpenTelemetrySubsystemDefinition.java

```diff
@@ -0,0 +1,159 @@
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(5000))
+            .build();
+
+    public static final SimpleAttributeDefinition MAX_QUEUE_SIZE = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.MAX_QUEUE_SIZE, ModelType.LONG, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(2048))
+            .build();
+
+    public static final SimpleAttributeDefinition MAX_EXPORT_BATCH_SIZE = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.MAX_EXPORT_BATCH_SIZE, ModelType.LONG, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(512))
+            .build();
+
+    public static final SimpleAttributeDefinition EXPORT_TIMEOUT = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.EXPORT_TIMEOUT, ModelType.LONG, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(30000))
+            .build();
+
+    public static final String GROUP_SAMPLER = "sampler";
+    public static final SimpleAttributeDefinition SAMPLER = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.SAMPLER_TYPE, ModelType.STRING, true)
+            .setAllowExpression(true)
+            .setXmlName(OpenTelemetryConfigurationConstants.TYPE)
+            .setAttributeGroup(GROUP_SAMPLER)
+            .setRestartAllServices()
+            .build();
+
+    public static final SimpleAttributeDefinition SAMPLER_ARG = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.SAMPLER_ARG, ModelType.DOUBLE, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SAMPLER)
+            .setRestartAllServices()
+            .build();
+
+    static final AttributeDefinition[] ATTRIBUTES = {
+            SERVICE_NAME, EXPORTER, ENDPOINT, SPAN_PROCESSOR_TYPE, BATCH_DELAY, MAX_QUEUE_SIZE, MAX_EXPORT_BATCH_SIZE,
+            EXPORT_TIMEOUT, SAMPLER, SAMPLER_ARG
+    };
+
+    public static final OpenTelemetrySubsystemDefinition INSTANCE = new OpenTelemetrySubsystemDefinition();
+
+    protected OpenTelemetrySubsystemDefinition() {
+
+        super(new SimpleResourceDefinition.Parameters(OpenTelemetrySubsystemExtension.SUBSYSTEM_PATH,
+                OpenTelemetrySubsystemExtension.getResourceDescriptionResolver())
+                .setAddHandler(OpenTelemetrySubsystemAdd.INSTANCE)
+                .setRemoveHandler(OpenTelemetrySubsystemRemove.INSTANCE));
```

Comment:
You need to register a capability and have it depend on the "org.wildfly.weld" capability. That's how we ensure this subsystem can't be added if the weld subsystem is not and is what tells Galleon to reject an attempt to generate a config that does that.

See org.wildfly.extension.microprofile.config.smallrye.MicroProfileSubsystemDefinition.